native-platform/src/test/groovy/net/rubygrapefruit/platform/file/PosixFilesTest.groovy

```diff
@@ -317,6 +317,53 @@ class PosixFilesTest extends FilesTest {
         e.message == "Could not read symlink $symlinkFile: could not readlink (errno 22: Invalid argument)"
     }
 
+    @Unroll
+    def "can get mode for a file behind symbolic link"() {
+        def testFile = new File(tmpDir.root, name)
+        testFile.parentFile.mkdirs()
+        testFile.text = "hi"
+        files.setMode(testFile, 0660)
+
+        def symlinkFile = new File(tmpDir.root, name + '.link')
+
+        when:
+        files.symlink(symlinkFile, testFile.name)
+
+        then:
+        files.getMode(symlinkFile, true) == files.getMode(testFile)
+        files.getMode(symlinkFile, false) != files.getMode(testFile)
+        files.getMode(symlinkFile, false) == files.getMode(symlinkFile)
+
+        files.getMode(testFile, true) == files.getMode(testFile)
+
+        where:
+        name << names
+    }
+
+    @Unroll
+    def "can get mode for a file behind broken symbolic link"() {
+        def brokenSymlinkFile = new File(tmpDir.root, name + '.link')
+        brokenSymlinkFile.parentFile.mkdirs()
+        files.symlink(brokenSymlinkFile, name)
+
+        when:
+        files.getMode(brokenSymlinkFile, false)
+        files.getMode(brokenSymlinkFile)
+
+        then:
+        noExceptionThrown()
+
+        when:
+        files.getMode(brokenSymlinkFile, true)
+
+        then:
+        noExceptionThrown()
```

Comment:
:x: This doesn't make sense to me. The caller asks to follow the symlink explicitly, but it is not possible to follow the symlink, because it's broken. This should throw an exception, just as if you asked for a non-existent file in the first place, not silently return the permissions for the referring symlink.